centaur/src/main/scala/centaur/test/Test.scala

```diff
@@ -780,13 +780,39 @@ object Operations extends StrictLogging {
       }
 
       def checkArchived(): IO[Boolean] = for {
-        archiveStatus <- CentaurCromwellClient.archiveStatus(workflowId)
+        archiveStatus <- CentaurCromwellClient.archiveStatus(submittedWorkflow.id)
         isArchived <- validateMetadataArchiveStatus(archiveStatus)
       } yield isArchived
 
+      def validateMetadataSourceArchived(): IO[Boolean]  = for {
+        metadataSource <- CentaurCromwellClient.metadataWithId(submittedWorkflow.id).map(_.asFlat.stringifyValues.get("metadataSource"))
+        isMetadataSourceArchived <- {
+          if (metadataSource == Option(JsString("Archived"))) {
+            IO.pure(true)
+          } else if (metadataSource == Option(JsString("Unarchived"))) {
+            IO.pure(false)
+          } else {
+            throw CentaurTestException(
+              s"`Metadata` endpoint returned unknown value for `metadataSource`: $metadataSource",
+              workflowDefinition,
+              submittedWorkflow
+            )
+          }
+        }
+      } yield isMetadataSourceArchived
+
       def eventuallyArchived(): IO[Unit] = {
         checkArchived() flatMap {
-          case true => IO.pure(())
+          case true =>
+            validateMetadataSourceArchived() flatMap {
+              case true => IO.pure(())
+              case false =>
+                throw CentaurTestException(
+                  "`Query` endpoint returns metadata status \"Archived\" but `metadata` endpoint returns metadata source \"Unarchived\"",
```

Comment:
Since metadata and the summary table are slightly disjoint, I wonder whether this is really an error, or just a possible temporary state that the system will move out of next time we ask?

In other words, if we get archived in one place and unarchived in another, maybe we just need to wait a few seconds and the system will be back in sync?

Reply:
I think it depends. Summary gets updated later than metadata, so:

| summary status | metadata status | is it ok |
| ---------------- | ---------------- | ------ |
| archived | unarchived | not ok |
| unarchived | archived | ok |

Reply:
I think it depends. Summary gets updated later than metadata, so:

| summary status | metadata status | is it ok |
| ---------------- | ---------------- | ------ |
| archived | unarchived | not ok |
| unarchived | archived | ok |